webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Details/Cards/Virtualization/VirtVmware.js

```diff
@@ -48,6 +48,41 @@ const VirtVmware = ({ vm }) => (
         {vm.corespersocket}
       </DescriptionListDescription>
     </DescriptionListGroup>
+    {Object.values(vm.disks).map((disk, index) => (
```

Comment:
I can't test the whole pr, but are `vm.disks` and `vm.partitions` always defined? if not this can crash the whole card, so maybe we should use `vm.disks || {}`

Reply:
Just tested it on a machine, on which no vmware tools are installed:

![vm-disk-no-tools](https://github.com/theforeman/foreman/assets/25485845/e20888cb-6a8c-409b-94e9-9987685c8b81)

With vmware tools, it look like:

![vm-disk-on](https://github.com/theforeman/foreman/assets/25485845/92d3bebd-a4ee-4704-bfbc-a2dde251d7f8)

So, it doesn't crash.